contracts/registries/ThirdPartyRegistry.sol

```diff
@@ -483,98 +503,74 @@ contract ThirdPartyRegistry is OwnableInitializable, NativeMetaTransaction {
     }
 
      /**
-    * @notice Consume third party slots
-    * @param _thirdPartyId - third party id
-    */
-    function _consumeSlots(
+     * @notice Review third parties with Merkle Root
+     * @dev The amount of slots should be the same as the amount of items in the merkle tree 
+     * @param _thirdPartyId - third party id
+     * @param _root - Merkle tree root
+     * @param _consumeSlotsParams - Data to consume slots mutilple times in a single transaction
+     */
+    function reviewThirdPartyWithRoot(
         string calldata _thirdPartyId,
-        uint256 _qty,
-        bytes32 _hash,
-        bytes32 sigR,
-        bytes32 sigS,
-        uint8 sigV
-    ) internal {
-        address sender = _msgSender();
-
-        ThirdParty storage thirdParty = thirdParties[_thirdPartyId];
-
-        require(_qty > 0, "TPR#consumeSlots: INVALID_QTY");
-        require(thirdParty.maxItems >= thirdParty.consumedSlots.add(_qty), "TPR#consumeSlots: NO_ITEM_SLOTS_AVAILABLE");
-
-        bytes32 messageHash = keccak256(address(this), _thirdPartyId, _qty, _hash);
-        require(_thirdParty.receipts[messageHash] > 0, "TPR#consumeSlots: MESSAGE_ALREADY_PROCESSED");
-
-        address signer = ecrecover(
-            messageHash,
-            sigV,
-            sigR,
-            sigS
-        );
-        require(thirdParty.managers[signer], "TPR#consumeSlots: INVALID_SIGNER");
-
-        thirdParty.receipts[messageHash] = _qty;
-        thirdParty.consumedSlots = thirdParty.consumedSlots.add(_qty);
-
-
-        emit ItemSlotConsumed(
-            _thirdPartyId,
-            _qty,
-            signer,
-            sender
-        );
-    }
-
-     /**
-    * @notice Review third party items by using a merklee tree root
-    * @param _thirdParties - Third parties with items to be reviewed
-    */
-    function reviewThirdPartiesWithRoot(string calldata _thirdPartyId,
         bytes32 _root,
-        uint256 _qty,
-        bytes32 sigR,
-        bytes32 sigS,
-        uint8 sigV
+        ConsumeSlotsParam[] calldata _consumeSlotsParams
     ) onlyCommittee external {
         address sender = _msgSender();
 
-        require(_root != bytes32(0), "TPR#reviewThirdPartiesWithRoot: INVALID_ROOT");
+        require(_root != bytes32(0), "TPR#reviewThirdPartyWithRoot: INVALID_ROOT");
 
         ThirdParty storage thirdParty = thirdParties[_thirdPartyId];
-        _checkThirdParty(thirdParty);
 
-        if (qty > 0) {
-            _consumeSlots(_thirdPartyId, _root, _qty, sigR, sigS, sigV);
-        }
+        _consumeSlots(_thirdPartyId, _consumeSlotsParams);
 
         thirdParty.isApproved = true;
         thirdParty.root = _root;
 
-        emit ThirdPartyReviewedWithRoot(thirdPartyReview.id, thirdPartyReview.root, thirdParty.isApproved, sender);
+        emit ThirdPartyReviewedWithRoot(_thirdPartyId, _root, thirdParty.isApproved, sender);
```

Comment:
I think that the `isApproved` is obsolete here. It will always be true. Maybe we can leave it as it is just to make events handlers optimal

Reply:
What do you mean by optimal?

Reply:
They don't need to have business logic like ``` If the event is a `ThirdPartyReviewedWithRoot` then I need to set the third party as approved``` by always receive that value in the event parameters we can just pass through it